### PR TITLE
VTypes is ready for review by team.

### PR DESCRIPTION
**Pull request by fancyElise**
Christine is the primary reviewer, but everyone else should feel free to review it for completeness/awareness.

Added a function called getSQLiteType to retrieve a string representation of the String, Double, Integer, and Boolean types.

made modifications to the VComponent update function that affected all of the other Big 4.  solution files also needed to incorporate file structure changes.

### CODE REVIEW

File: VAST/h/VType.h [hunk 1]
```diff
@@ -70,80 +101,285 @@ class String : public VType
...
+
+		if (this != &s)   // only copy if the object passed in is not already this one 
+		{
+			this->type = s.getType();
+			this->stringValue = s.s_value();
+			this->val = s.s_value();
+		}
+		return this;     // return the object  
+	}
+///*Creates output stream for String data.*/
```
Comment (ceodenwald) [inline VAST/h/VType.h:345]: Commented out code

File: VAST/h/VType.h [hunk 2]
```diff
@@ -157,92 +393,148 @@ class Double : public VType
...
-		VType::type = DOUBLE_TYPE;*/
+		val = 0.0;
+		VType::stringValue = "0.0";
+		VType::type = DOUBLE_TYPE;
 	}
 
-	Double(VType& obj) : VType()
-	{
-
+	Double(VType* obj) : VType()
```
Comment (ceodenwald) [inline VAST/h/VType.h:405]: No description on constructor

File: VAST/h/VType.h [hunk 3]
```diff
@@ -157,92 +393,148 @@ class Double : public VType
...
 	/* companion to the operator overload for + below*/
 	double operator+=(const double rhs)
 	{
-		return 0.0;
+		val += rhs;
+		return val;
 	};
 	
-	double operator+=(const Double& rhs)
+	double operator+=(Double& rhs)
```
Comment (ceodenwald) [inline VAST/h/VType.h:479]: No description/comment

File: VAST/h/VType.h [hunk 4]
```diff
@@ -157,92 +393,148 @@ class Double : public VType
...
 	{
-		return 0.0;
+		val += rhs.value();
+		return val;
 	};
 
-	double Double::operator -=(const Double& otherDouble)
+	double Double::operator -=(Double& otherDouble)
```
Comment (ceodenwald) [inline VAST/h/VType.h:487]: No description/comment

File: VAST/h/VType.h [hunk 5]
```diff
@@ -258,92 +550,176 @@ class Integer : public VType
...
-	};
+		val = 0;
+		VType::type = INTEGER_TYPE;
+		VType::stringValue = "0";
 
-	Integer(VType& obj) : VType()
-	{
+	};
 
+	Integer(VType* obj) : VType()//VType(Integer::stringValue)
```
Comment (ceodenwald) [inline VAST/h/VType.h:574]: No description

File: VAST/h/VType.h [hunk 6]
```diff
@@ -258,92 +550,176 @@ class Integer : public VType
...
-	int operator+=(const Integer& rhs)
+	int operator+=(Integer& rhs)
 	{
-		return 0;
+		val += rhs.value();
+		return val;
 	};
 
-	int Integer::operator -=(const Integer& otherInteger)
+	int Integer::operator -=(Integer& otherInteger)
```
Comment (ceodenwald) [inline VAST/h/VType.h:672]: No comment/description

File: VAST/h/VType.h [hunk 7]
```diff
@@ -359,83 +735,153 @@ class Boolean : public VType
...
-		VType::type = BOOLEAN_TYPE;*/
+		val = false;
+		VType::type = BOOLEAN_TYPE;
+		VType::stringValue = "false";
 	};
 
-	Boolean(VType& obj) : VType()
-	{
-
+	Boolean(VType* obj) : VType()
```
Comment (ceodenwald) [inline VAST/h/VType.h:767]: No description

File: VAST/h/VType.h [hunk 8]
```diff
@@ -448,61 +894,170 @@ class Vector3 : public VType
...
 		_z = 0;
-		VType::type = VECTOR3_TYPE;*/
+		VType::type = VECTOR3_TYPE;
+		stringstream ss;
+		ss << _x << "," << _y << "," << _z;
+		VType::stringValue = ss.str();
 	}
 
-	Vector3(VType& obj) : VType()
+	Vector3(VType* obj) : VType()
```
Comment (ceodenwald) [inline VAST/h/VType.h:1008]: No description

File: VAST/h/VType.h [hunk 9]
```diff
@@ -513,73 +1068,315 @@ variable public to users of VType.h.
 class Array : public VType
 {
 private:
-	string val; // concatenated string form of all values
 	string stowType;
 	int size; 
-	VType* array;
-
+	VType** stowArray;
```
Comment (ceodenwald) [inline VAST/h/VType.h]: no description

File: VAST/h/VType.h [hunk 10]
```diff
@@ -589,74 +1386,265 @@ class Array : public VType
...
-	/* Sets the type of the stowed VTypes in the Array and returns true 
-	if this was able to be accomplished; is good for converting numbers 
-	and strings back and forth.  SetStowType conversions will not delete 
-	the old stowed objects in favor of the new VType.
-	Will return false and set will fail if types cannot be converted.  */
-	bool setStowType(string type)
-	{
-		return false;
-	}
+	///* Sets the type of the stowed VTypes in the Array and returns true
```
Comment (ceodenwald) [inline VAST/h/VType.h:1487]: Commented out code

File: VAST/h/VType.h [hunk 11]
```diff
@@ -4,12 +4,15 @@
 #include <string>
```
Comment (ceodenwald) [inline VAST/h/VType.h:4]: Overall, very good... would suggest writing pass/fail criteria/descriptions for each test listed

**Review by ceodenwald — CHANGES_REQUESTED** [review]
Please see the comments in the code

File: VAST/h/VType.h (hunk 9 above)
Comment (fancyElise) [reply VAST/h/VType.h]: none required...but I'll put some

File: VAST/h/VType.h (hunk 11 above)
Comment (fancyElise) [reply VAST/h/VType.h:4]: good idea

File: VAST/h/VType.h (hunk 1 above)
Comment (fancyElise) [reply VAST/h/VType.h:345]: ok this change has been made

**Pull request merged**